java/arcs/core/util/MutableBiMap.kt

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.core.util
+
+import kotlinx.coroutines.sync.Mutex
+
+/**
+ * Thread-safe mutable bi-directional map utility class.
+ */
+class MutableBiMap<L, R>() {
```

Comment:
This seems like a good candidate to have implementing `MutableMap<L, R>`. Since it's essentially a regular map, with an added restriction that it be one-to-one and onto (invertable), which gives it the added performance boost of being able to do O(1) lookups of values' keys.

Reply:
I was mistaken in my understanding of this! Whoops!